driver-core/src/main/com/mongodb/internal/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -188,7 +188,7 @@ private void resumeableOperation(final AsyncBlock asyncBlock, final SingleResult
                 try {
                     List<T> convertedResults;
                     try {
-                        convertedResults = convertAndProduceLastId(result, changeStreamOperation.getDecoder(),
+                        convertedResults = convertAndProduceLastId(assertNotNull(result), changeStreamOperation.getDecoder(),
```

Comment:
Note: As per the changes in the AsyncSingleBatchCursor - async batch cursors no longer return null.